Simulator/Simulator/Experiments/src/networks.c

```diff
@@ -24,16 +24,107 @@ Network* network_3_L_3_3_3() {
 }
 
 
-Network* network_synfire_chain_10_layers(float connectivity, float s_strength) {
+Network* network_synfire_chain_n_layers(uint32_t n_layers, float connectivity, float amplitude, float s_strength, BOOL use_refract) {
 	// create network
 	Network* net = network_create();
 	network_add_neuron_class(net, neuron_class_create("LIF_NEURON", LIF_NEURON));
 	network_add_neuron_class(net, neuron_class_create("LIF_NEURON_REFRAC", LIF_REFRACTORY_NEURON));
 
 	// excitation and inhibition
-	network_add_synapse_class(net, synapse_class_create("GABA_A", -90.0f, 1.0f, 6, 10, VOLTAGE_DEPENDENT_SYNAPSE, 1));
-	network_add_synapse_class(net, synapse_class_create("AMPA", 0.0f, 1.0f, 3, 10, VOLTAGE_DEPENDENT_SYNAPSE, 1));
-	network_add_synapse_class(net, synapse_class_create("CONDUCTANCE_10_TAU", 0.0, 1.0f, 10, 10, CONDUCTANCE_SYNAPSE, 1));
+	network_add_synapse_class(net, synapse_class_create("GABA_A", -90.0f, amplitude, 6, 10, VOLTAGE_DEPENDENT_SYNAPSE, 1)); // inhibition
+	network_add_synapse_class(net, synapse_class_create("AMPA", 0.0f, amplitude, 1, 10, VOLTAGE_DEPENDENT_SYNAPSE, 1));	// excitation
+	network_add_synapse_class(net, synapse_class_create("CONDUCTANCE_10_TAU", 0.0, amplitude, 10, 10, CONDUCTANCE_SYNAPSE, 1));
+
+	NeuronClass* LIF_n_class = network_get_neuron_class_by_name(net, "LIF_NEURON");
+	NeuronClass* LIF_refrac_n_class = network_get_neuron_class_by_name(net, "LIF_NEURON_REFRAC");
+
+	SynapseClass* conductance_s_class = network_get_synapse_class_by_name(net, "CONDUCTANCE_10_TAU");
+	SynapseClass* inhibitory_s_class = network_get_synapse_class_by_name(net, "GABA_A");
+	SynapseClass* excitatory_s_class = network_get_synapse_class_by_name(net, "AMPA");
```

Comment:
Neuron and Synapse Classes can also be putted in the parameters and the big structure of parameters in the above function

Reply:
Currently is hard to do this and its more convenient this way